examples/hybrid/sphere/hadley_circulation.jl

```diff
@@ -0,0 +1,296 @@
+const κ₄ = 1.0e16          # hyperviscosity
+
+# time constants
+T = 86400.0 * 1.0
+dt = 60.0 * 60.0
+
+# set up 3D domain
+hv_center_space, hv_face_space = sphere_3D()
+
+# initial conditions
+coords = Fields.coordinate_field(hv_center_space)
+face_coords = Fields.coordinate_field(hv_face_space)
+
+p(z) = p₀ * exp(-z / H)
+ρ_ref(z) = p(z) / R_d / T₀
+
+y0 = map(coords) do coord
+    z = coord.z
+
+    z₀ = 0.5 * (z₂ + z₁)
+    if z > z₁ && z < z₂
+        q1 = 0.5 * (1.0 + cos(2pi * (z - z₀) / (z₂ - z₁)))
+    else
+        q1 = 0.0
+    end
+
+    ρq1 = ρ_ref(z) * q1
+
+    return (ρ = ρ_ref(z), ρq1 = ρq1)
+end
+
+# IC FieldVector
+y0 = Fields.FieldVector(ρ = y0.ρ, ρq1 = y0.ρq1)
+
+ystar = copy(y0)
+parameters = (;
+    coords = coords,
+    face_coords = face_coords,
+    τ = τ,
+    u₀ = u₀,
+    ω₀ = ω₀,
+    K = K,
+    ρ₀ = ρ₀,
+)
+
+# Set up the initial flow
+rhs!(ystar, y0, parameters, 0.0, dt, 1)
+
+# run!
+prob = ODEProblem(IncrementingODEFunction(rhs!), copy(y0), (0.0, T), parameters)
+sol = solve(
+    prob,
+    SSPRK33ShuOsher(),
+    dt = dt,
+    saveat = dt,
+    progress = true,
+    adaptive = false,
+    progress_message = (dt, u, p, t) -> t,
+)
+
+q1_error =
+    norm(sol.u[end].ρq1 ./ ρ_ref.(coords.z) .- y0.ρq1 ./ ρ_ref.(coords.z)) /
+    norm(y0.ρq1 ./ ρ_ref.(coords.z))
+@test q1_error ≈ 0.0 atol = 0.05
+
+Plots.png(
+    Plots.plot(
+        sol.u[trunc(Int, end / 2)].ρq1 ./ ρ_ref.(coords.z),
+        level = 11,
+        clim = (-0.1, 3.5),
+    ),
+    joinpath(path, "q1_half_day_level_11.png"),
+)
+
+anim = Plots.@animate for u in sol.u
+    Plots.plot(u.ρq1 ./ ρ_ref.(coords.z), level = 11, clim = (-0.1, 3.5))
+end
+Plots.mp4(anim, joinpath(path, "q1_anim_level_11.mp4"), fps = 10)
+
+# Remapping infrastructure to have lat-altitude Plots
+Nq = Spaces.Quadratures.degrees_of_freedom(
+    Spaces.quadrature_style(hv_center_space),
+)
+datafile_cc = "hadley_circulation.nc"
+NCDataset(datafile_cc, "c") do nc
+    # defines the appropriate dimensions and variables for a space coordinate
+    def_space_coord(nc, hv_center_space, type = "cgll")
+    def_space_coord(nc, hv_face_space, type = "cgll")
+    # defines the appropriate dimensions and variables for a time coordinate (by default, unlimited size)
+    nc_time = def_time_coord(nc)
+
+    # define variables
+    nc_rho = defVar(nc, "rho", Float64, hv_center_space, ("time",))
+    nc_q1 = defVar(nc, "q1", Float64, hv_center_space, ("time",))
+
+    # write data to netcdf file
+    for i in 1:length(sol.u)
+        nc_time[i] = sol.t[i]
+
+        # write fields to file
+        nc_rho[:, i] = sol.u[i].ρ
+        nc_q1[:, i] = sol.u[i].ρq1 ./ sol.u[i].ρ
+    end
+end
+
+# write out our cubed sphere mesh
+meshfile_cc = "mesh_cubedsphere.g"
+write_exodus(meshfile_cc, hv_center_space.horizontal_space.topology)
+
+# write out RLL mesh
+nlat = 90
+nlon = 180
+meshfile_rll = "mesh_rll.g"
+rll_mesh(meshfile_rll; nlat = nlat, nlon = nlon)
```

Comment:
@simonbyrne I added the remapping infrastructure in this example to be able to slice and plot lat-z data. I followed the example docs (now updated in #896 ) and how this is done in ClimaAtmos. 

When read, the `z` data looks fine, but the `lat` and `lon` sizes look off and the data not monotonic (their sizes should be 90 and 180 long, respectively). See: 
```
julia> lat = ncread("/home/valeria/CliMA/ClimaCore.jl/hadley_circulation.nc", "lat")
3458-element Vector{Float64}:
 -35.264389682754654
 -36.362523305127056
 -38.31776951931289
 -40.06731669533083
 -40.89339464913091
 -33.020430140278734
   ⋮
 -42.45046889871875
 -39.78669446992532
 -41.772505399580076
 -39.78669446992532
 -37.58860988580911

julia> lon = ncread("/home/valeria/CliMA/ClimaCore.jl/hadley_circulation.nc", "lon")
3458-element Vector{Float64}:
 -45.0
 -42.586678527746805
 -37.79054843599655
 -32.74314423305398
 -29.999999999999993
 -45.0
   ⋮
  45.00000000000001
  49.83280190367406
  35.028968545026046
  40.16719809632595
  45.0

julia> z = ncread("/home/valeria/CliMA/ClimaCore.jl/hadley_circulation.nc", "z")
30-element Vector{Float64}:
   200.0
   600.0
  1000.0
  1400.0
  1800.0
  2200.0
     ⋮
 10200.0
 10600.0
 11000.0
 11400.0
 11800.0
```
Could you please provide any advice on why this remapped data is wrong for `lat` and `lon`?

Reply:
I think this nc file is the intermediate file that is not remapped yet.

Reply:
Thank you so much Jia! You were totally right. I fixed it.